test/unit/throw.spec.js

```diff
@@ -26,6 +26,25 @@ describe('a test that throws', function () {
     });
   });
 
+  describe('extensible', function () {
+    it('should not crash if throwing non-extensible type', function (done) {
+      /* eslint-disable-next-line no-unused-vars */
+      var test = new Test('im async and throw string async', function (done2) {
+        process.nextTick(function () {
+          throw 'error';
+        });
+      });
+      suite.addTest(test);
```

Comment:
Possibly outside the scope, but rather than repeating [these lines](https://github.com/mochajs/mocha/blob/cc2bdf6c73f8e333f09fec83465bc4130ac8371a/test/unit/throw.spec.js#L37-L44) over and over, why don't we just use a function `runTest(test, done)` instead?